shepherd/storage/minio_storage.py

```diff
@@ -0,0 +1,38 @@
+from io import BytesIO
+
+from minio import Minio
+
+from shepherd.constants import DONE_FILE, ERROR_FILE
+from shepherd.storage.storage import Storage
+from shepherd.utils import minio_object_exists, pull_minio_bucket, push_minio_bucket
+
+
+class MinioStorage(Storage):
+    def __init__(self, minio: Minio):
+        self._minio = minio
+
+    def is_accessible(self) -> bool:
+        try:
+            self._minio.list_buckets()
+            return True
+        except BaseException:
+            return False
+
+    def pull_job_data(self, job_id: str, target_directory: str):
+        pull_minio_bucket(self._minio, job_id, target_directory)
+
+    def push_job_data(self, job_id: str, source_directory: str):
+        push_minio_bucket(self._minio, job_id, source_directory)
+
+    def job_failed(self, job_id: str, message: str):
```

Comment:
A better name would be imo `report_job_failed`

Reply:
Sure, what the hell.